python/ray/tune/impl/tuner_internal.py

```diff
@@ -150,7 +150,11 @@ def _expected_utilization(self, cpus_per_trial, cpus_total):
             concurrent_trials = math.inf
 
         actual_concurrency = min(
-            (cpus_total // cpus_per_trial, num_samples, concurrent_trials)
+            (
+                (cpus_total // cpus_per_trial) if cpus_per_trial else 0,
```

Comment:
nit: Why not just terminate earlier?
```python
if cpus_per_trial == 0:
    return 0
```

Reply:
Either works! Though this got already merged into master in this state, so I'd not want to modify it here